the-collab-lab/tcl-15-smart-shopping-list
# #issue-13: View details of the list items

## Type of Changes

|      | Type                       |
| --- | -------------------------- |
|  ✓    | :sparkles: New feature     |
|  ✓    | :bug: Bug fix              |

Closes #13 

## Acceptance Criteria
- [X]  From the shopping list, the user can tap to navigate to a detailed view of the item
- [X] Detail view includes the following information:
      * Name of the item
      * Last purchased date
      * Estimated number of days until the next purchase
      * Number of times the item has been purchased
- [X]  When in the “item details” view, a back arrow should appear in the header that takes the user back to the “list” view.

## Description
**This PR adds a details-view to the list items and fixes semantic errors in the previous code**

**First:** It creates `ListItemDetails` which is a component responsible for showing the user the details of an item once the user clicks the info button associated with that item.

**Second:** It adds three new functions to `helper.js`:
   * `getDaysSinceLastPurchase`: Gets the number of days since the last purchase.
   * `calculateNextPurchase`: Calculates the estimated date for the next purchase.
  * `getDaysUntilNextPurchase`: Gets the number of estimated days until the next purchase.

**Third**: It fixes the way the items are colored and sorted on the list.

## Update
A user can now view the details of the items on their list.
## How things look so far
![the details-view of the list items](https://i.ibb.co/qxKYB85/detailsview.gif)

## Testing Steps / QA Criteria

* From your terminal, pull down this branch with `git pull origin se-hn-issue-13` 
* Check that branch out with `git checkout hn-yk-issue-13`
* Run `yarn` and `yarn start` to run the [local server](http://localhost:3000/).
* Join a list by using the **"wool haifa muong"** token, or create your own list and add some items to it.
* Click the info button to an item right next to the trash icon and a modal should pop up to show you the details of the item. If you've never purchased that item before a **No recorded purchases yet** message should show. Otherwise, some info about that item should show.  

**Note**: In case you needed to, please use [this converter](https://datetomilliseconds.netlify.app/) to convert dates to milliseconds.

**Thank you so much for your time.**
